src/queue.js

```diff
@@ -1,38 +1,51 @@
 export default class Queue {
   constructor() {
-    this.dataStore = [];
-    this.enqueue = enqueue;
-    this.dequeue = dequeue;
-    this.front = front;
-    this.back = back;
-    this.toString = toString;
-    this.empty
+    this.arr = [];
+    this.total = 0;
 }
-
-//
   enqueue(element) {
-
+    this.arr[this.total++] = element
   }
 
   dequeue() {
+    if(this.total == null) {
+      return null
+    }
+   while( this.arr.shift() ) {
+      this.arr[--this.total]
+    }
+   return this.arr
+  }
 
+  length() {
+    return this.total
   }
 
   front() {
-
+    if(this.total = 1 && this.total == null) {
+      return null
+    }else{
+      this.arr.shift()
+      return this.arr
+    }
   }
 
   back() {
+    if(this.total-1 == null) {
+      return null
+    }else{
+      this.arr.pop()
+      return this.arr
+    }
   }
 
-  toString() {
-  }
-
-  push() {
-
-  }
-  peek() {
 
+  isEmpty() {
+    if(this.total > 0) {
```

Comment:
returning a statement that checks for something will return true or false based on whether the condition was passed or not.